var/spack/repos/builtin/packages/qthreads/restrict.patch

```diff
@@ -0,0 +1,12 @@
+--- a/include/qthread/common.h.in
++++ b/include/qthread/common.h.in
+@@ -84,7 +84,9 @@
+ /* Define to the equivalent of the C99 'restrict' keyword, or to
+    nothing if this is not supported.  Do not define if restrict is
+    supported directly.  */
++#ifndef restrict
```

Comment:
This is a `config.h.in` file for autoconf. Configure will replace the `#undef` with `#define`, depending on what it detects for the platform. If another package does the same, then `restrict` will be defined twice. Hence this (potential) `#define` needs to be protected by an `#ifdef`.